torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp

```diff
@@ -110,7 +110,12 @@ std::shared_ptr<PointwiseParams> getPointwiseHeuristics(
         (int64_t)dataTypeSize(inp->getDataType().value(), index_type));
   }
 
-  auto ref_root = largest_out->getMaybeRFactorDomain();
+  // We always cacheBefore output at the beginning of the scheduling. And after
```

Comment:
Is this the fix for trivial reductions? Could you show an example with trivial reductions?

Reply:
For example, if you have:
```
T0[b0, I1, I2] = input
T1[r0, I1, I2] = sum(T0)
output = T1
```
Then when computing the heuristics, all positions you are using are in `T1[r0, I1, I2]`. But during scheduling, you will do a `T1.cacheBefore()` and get
```
T0[b0, I1, I2] = input
T1[r0, I1, I2] = sum(T0)
T2[I1, I2] = set(T1)
output = T2
```
so all the positions you are referring to are now in `T2[I1, I2]`, which is 1 less than in `T1[r0, I1, I2]`. And this is another example why I hate trivial reduction and want to replace it with squeeze :).